build error referencing /usr/lib/libpython with conda/mac

I tried to walk through the steps to try it from a conda environment, but ran into an error, on mac:

```
poetry run pytest
============================================== test session starts ===============================================
platform darwin -- Python 3.11.5, pytest-7.4.2, pluggy-1.3.0
rootdir: /private/var/tmp/poetry-test/poetry-demo
plugins: ziggy-pydust-0.2.12
collected 1 item / 1 error

# ,,,

.../opt/miniconda3/envs/poetry.3.11/lib/python3.11/site-packages/_pytest/runner.py:341: in from_call
    result: Optional[TResult] = func()
.../opt/miniconda3/envs/poetry.3.11/lib/python3.11/site-packages/_pytest/runner.py:372: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
.../opt/miniconda3/envs/poetry.3.11/lib/python3.11/site-packages/pydust/pytest_plugin.py:62: in collect
    h = TestProtocol.Header.unpack(proc.stdout)
.../opt/miniconda3/envs/poetry.3.11/lib/python3.11/site-packages/pydust/pytest_plugin.py:230: in unpack
    (tag, bytes_len) = struct.unpack("<II", buffer.read(8))
E   struct.error: unpack requires a buffer of 8 bytes
------------------------------------------------ Captured stderr -------------------------------------------------
dyld[32874]: Library not loaded: @rpath/libpython3.11.dylib
  Referenced from: <98F79F69-BB8F-365C-8D5A-E3C2161D139D> /private/var/tmp/poetry-test/poetry-demo/zig-out/bin/hello.test.bin
  Reason: tried: '/usr/local/lib/libpython3.11.dylib' (no such file), '/usr/lib/libpython3.11.dylib' (no such file, not in dyld cache)
```

Steps:

```zsh
conda create -n poetry.3.11 python=3.11
conda activate --stack poetry.3.11
python -m pip install poetry pytest
poetry new poetry-demo
cd !$
poetry add -G dev ziggy-pydust
# add files and stuff
poetry run pytest
```

--- a/pydust/buildzig.py
+++ b/pydust/buildzig.py
@@ -13,6 +13,7 @@ limitations under the License.
 """
 
 import contextlib
+import os
 import subprocess
 import sys
 import sysconfig
@@ -24,7 +25,12 @@ from pydust import config
 PYVER_MINOR = ".".join(str(v) for v in sys.version_info[:2])
 PYVER_HEX = f"{sys.hexversion:#010x}"
 PYINC = sysconfig.get_path("include")
-PYLIB = sysconfig.get_config_var("LIBDIR")
+PYLIBDIR = sysconfig.get_config_var("LIBDIR")
+PYLDLIB = sysconfig.get_config_var("LDLIBRARY")
+
+# Strip libpython3.11.a.so => python3.11.a
+PYLDLIB = PYLDLIB[3:] if PYLDLIB.startswith("lib") else PYLDLIB
+PYLDLIB = os.path.splitext(PYLDLIB)[0]
 
 
 def zig_build(argv: list[str]):
@@ -210,8 +216,8 @@ def generate_build_zig(build_zig_file):
                 pydust: []const u8, compile: *std.Build.CompileStep, pyconf: *std.Build.Step.Options
             ) void {{
                 configurePythonInclude(pydust, compile, pyconf);
-                compile.linkSystemLibrary("python{PYVER_MINOR}");
-                compile.addLibraryPath(.{{ .path =  "{PYLIB}" }});
+                compile.linkSystemLibrary("{PYLDLIB}");
+                compile.addLibraryPath(.{{ .path =  "{PYLIBDIR}" }});
             }}
             """
         )

